eng/tools/npm-helpers.ps1

```diff
@@ -0,0 +1,162 @@
+$EngPath = Resolve-Path "${PSScriptRoot}/.."
+$Engcommon = Join-Path $EngPath "common"
+$EngCommonScriptsPath = Join-Path $Engcommon "scripts"
+. (Join-Path $EngCommonScriptsPath common.ps1)
+
+function GetNpmTagVersions($packageName)
+{
+  try
+  {
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${PackageName}"
+    $latest = ($existingVersion."dist-tags").latest
+    $next = ($existingVersion."dist-tags").next
+    Write-Host "Latest version: $latest"
+    Write-Host "Next version: $next"
+    if ($latest -eq $null) {
+      Write-Host "'latest' tag is not present in npm registry for package $packageName"
+    }
+    if ($next -eq $null) {
+      Write-Host "'Next' tag is not present in npm registry for package $packageName"
+    }
+
+    return New-Object PSObject -Property @{
+      latest = $latest
+      next = $next
+    }
+  }
+  catch
+  {
+    return $null
+  }
+}
+
+function GetNpmPackageVersions ($packageName)
+{
+  try
+  {
+    Write-Host "Checking versions present on npm for package $packageName"
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${packageName}"
```

Comment:
Do we need to consider filtering these results. There are a lot of versions here (183 so far). I wonder if NPM will start filtering what it gives back to us, or whether the list will grow out of control.

Reply:
I don't see any filter option in API itself. But we are filtering out all alpha and dev versions from actual usage. I will check and update if there is any filer for versions.